### DISCUSSION TITLE
PluginServer - uninstall vs undeploy

### DISCUSSION THREAD

**Discussion by safisa** [eclipse-theia/theia#13378] [Q&A]
Hi,

The PluginServer interface contains those declarations:
uninstall(pluginId: PluginIdentifiers.VersionedId)
undeploy(pluginId: PluginIdentifiers.VersionedId)

What are the differences between those methods?
I know that the uninstall is called when uninstalling an extension, but when is the undeploy is used?

Thanks

**Comment by xai** [comment]
Hi @safisa!

I investigated this and did not find a call to undeploy(). Setting breakpoints also had no effect whatsoever I did with an extension. My best guess is that it is some kind of legacy function, and most of what it did/does is now handled by uninstall().

@tsmaeder has recently shared some thoughts on improving this part of the code base to clarify and decouple the actions that are performed on plugins: #13416
